tests/integration/targets/opensearch/tasks/test_vpc_setup.yml

```diff
@@ -0,0 +1,134 @@
+    - name: gather AZ info in VPC for test purpose
+      amazon.aws.aws_az_info:
+        region: "{{ aws_region }}"
+      register: az_info
+    - assert:
+        that:
+          # We cannot run the test if this region does not have at least 2 AZs
+          - "az_info.availability_zones | length >= 2"
+    - set_fact:
+        test_az_zone1: "{{ az_info.availability_zones[0].zone_name }}"
+        test_az_zone2: "{{ az_info.availability_zones[1].zone_name }}"
+    - name: Create subnet_1 for use in testing
+      ec2_vpc_subnet:
+        state: present
+        vpc_id: "{{ testing_vpc.vpc.id }}"
+        cidr: 10.55.77.0/24
+        az: "{{ test_az_zone1 }}"
+        resource_tags:
+          Name: "{{ tiny_prefix }}-subnet_1"
+          AnsibleTest: AnsibleTestVpc
+        wait: true
+      register: testing_subnet_1
+
+    - name: Create subnet_2 for use in testing
+      ec2_vpc_subnet:
+        state: present
+        vpc_id: "{{ testing_vpc.vpc.id }}"
+        cidr: 10.55.78.0/24
+        az: "{{ test_az_zone2 }}"
+        resource_tags:
+          Name: "{{ tiny_prefix }}-subnet_2"
+          AnsibleTest: AnsibleTestVpc
+        wait: true
+      register: testing_subnet_2
+
+    - name: Create routing rules
+      ec2_vpc_route_table:
+        vpc_id: "{{ testing_vpc.vpc.id }}"
+        routes:
+          - dest: 0.0.0.0/0
+            gateway_id: "{{ igw.gateway_id }}"
+        subnets:
+          - "{{ testing_subnet_1.subnet.id }}"
+          - "{{ testing_subnet_2.subnet.id }}"
+        tags:
+          Name: "{{ tiny_prefix }}-route"
+          AnsibleTest: AnsibleTestVpc
+
+    - name: Create security group for use in testing
+      ec2_group:
+        name: "{{ tiny_prefix }}-sg"
+        description: a security group for ansible tests
+        vpc_id: "{{ testing_vpc.vpc.id }}"
+        tags:
+          Name: "{{ tiny_prefix }}-sg"
+          AnsibleTest: AnsibleTestVpc
+        rules:
+          - proto: tcp
+            from_port: 22
+            to_port: 22
+            cidr_ip: 0.0.0.0/0
+          - proto: tcp
+            from_port: 80
+            to_port: 80
+            cidr_ip: 0.0.0.0/0
+      register: sg
+
+    # A private route53 zone is needed to configure a custom endpoint in the OpenSearch cluster.
+    # See https://docs.aws.amazon.com/opensearch-service/latest/developerguide/customendpoint.html
+    - name: Create private Route53 zone for the VPC
+      register: route53_zone
+      route53_zone:
+        comment: "zone for ansible integration tests"
+        zone: "ansible-integ-test.com"
+        state: present
+        vpc_id: '{{ testing_vpc.vpc.id }}'
+        tags:
+          Name: "{{ tiny_prefix }}-zone"
+          AnsibleTest: AnsibleTestVpc
+
+    - name: Set fact for route53 zone id
+      set_fact:
+        route53_zone_id: "{{ route53_zone.result.zone_id }}"
+
+    - name: Create KMS key for test purpose
+      # The key is needed for OpenSearch encryption at rest.
+      aws_kms:
```

Comment:
last thing.  
the aws_kms key should also be cleaned up.

Reply:
This is already done here: https://github.com/ansible-collections/community.aws/pull/859/files#diff-0a2082f73b60bcc897b26905fabe43ec4ab77ced8f016f27014ae7e3bbbf8078R45-R56